conn.go

```diff
@@ -347,6 +376,97 @@ func (conn *Conn) inWorker() {
 	}
 }
 
+// sigWorker runs in an own goroutine, getting signals and handling
+// them appropriately.
+func (conn *Conn) sigWorker() {
+WorkerLoop:
```

Comment:
This label would be unnecessary if you just `return` instead of `break` and drop the `continue` below.